src/mds/MDSDaemon.cc

```diff
@@ -144,21 +144,25 @@ void MDSDaemon::asok_command(
   } else if (command == "exit") {
     outbl.append("Exiting...\n");
     r = 0;
-    std::thread t([this](){
+    auto t = new std::thread([this](){
 		    // Wait a little to improve chances of caller getting
 		    // our response before seeing us disappear from mdsmap
 		    sleep(1);
+		    std::lock_guard l(mds_lock);
 		    suicide();
 		  });
+    (void)t;
```

Comment:
Would it not be better to do:

`t.detach();` instead?

Reply:
I think ``auto t`` should make ``t`` a ``std::thread*``, not a smart pointer?

Reply:
oh, std::thread::detach(), gotcha